src/pydrex/core.py

```diff
@@ -234,28 +272,93 @@ def derivatives(
     Returns a tuple with the rotation rates and grain volume fraction changes.
 
     """
-    # Based on subroutine DERIV in original Fortran.
-    strain_energies = np.empty(n_grains)
-    orientations_diff = np.empty((n_grains, 3, 3))
-    for grain_index in range(n_grains):
-        orientation_change, strain_energy = _get_rotation_and_strain(
-            phase,
-            fabric,
-            orientations[grain_index],
-            strain_rate,
-            velocity_gradient,
-            stress_exponent,
-            deformation_exponent,
-            nucleation_efficiency,
+    if regime == DeformationRegime.min_viscosity:
+        # Do absolutely nothing, all derivatives are zero.
+        # TODO: Consider resetting the texture to isotropic, treat it like a phase
+        # change or something?
```

Comment:
Agreed, the tracker idea seems valuable. I had made a conscious effort to limit the amount of things tracked on `Mineral`s to ensure loose coupling (and avoid excessive memory use), but a history of regimes seems sensible. Instead of an array of values for each timestep, which would end up fairly sparse in most models (matrix_dislocation most of the time) I could do an array of tuples of (regime, duration in timesteps).